### ISSUE TITLE
Separate vanilla/mod commands from Sponge commands in implementation

### ISSUE THREAD

**Issue by dualspiral** [SpongePowered/Sponge#3594]
Right now, we have a weird mesh of Sponge parameterised commands being on the vanilla Brig tree, but other commands not being. This weird inconsistency has incurred maintenance penalties and means that Sponge managed commands may be managed in totally different ways.

I plan to split each registrar into its own Brig tree (or no Brig tree on the server, with hooks enabling behaviours that might otherwise traverse the tree). The client will still receive a merged tree. Plugins won't notice a difference, as the API won't change. Mods that try to traverse the brig tree may do, we need to think about that.

We also need to make sure that `/execute` and `/help`, amongst other commands, continue to work correctly.

This will block some nice-to-haves on commands, including the possiblility of dynamic addition/removal of Sponge commands.

(I plan to work on this when I get a bit of time, but this issue is here to be able to reference issues that might be blocked by this work)
